internal/pkg/config/matcher_test.go

```diff
@@ -296,6 +296,38 @@ Field: qux`,
 	}
 }
 
+func TestFieldTagMatcherUnmarshalling(t *testing.T) {
+	testCases := []struct {
+		desc, yaml        string
+		shouldErrorOnLoad bool
+	}{
+		{
+			desc: "unknown field is not allowed",
+			yaml: `
+Key: good
+UnknownField: bad`,
+			shouldErrorOnLoad: true,
+		}, {
+			desc: "valid field tag config",
+			yaml: `
+Key: good
+Value: AlsoGood`,
+			shouldErrorOnLoad: true,
```

Comment:
This case is supposed to be valid, so there shouldn't be an error, right?

Reply:
Well, in the config file you used "val", not "Value" lol
Let me know if you want to support all four: "val", "Val", "value", "Value".

Reply:
Sorry for not being clearer. The test name is "valid field tag config", so I would expect this case to test for success. I think it's good to have a "success" case. Otherwise, this case seems redundant with the previous one.

Reply:
Consider using lowercase keys in the test to validate that these are accepted.

Reply:
I agree that this seems counter-intuitive.  Probably the biggest issue here is that `Value: AlsoGood` really suggests that the value is correct.  If I were to leave it how it is, I'd write it as `Value: ShouldBeVal`.

Either here or in a future PR, I would say that we should accept a `Value` here as well.  At which point, I might update this test case to be something like

```
Key: good
Val: AlsoGood
Value: BadForRedundancy
```